### ISSUE TITLE
allow for api to be an object

### ISSUE THREAD

**Issue by dcharbonnier** [krakenjs/swaggerize-hapi#102]
Forcing the api to be a file path is inconvenient, I personally split my api to different files and dynamically generate an api, this force me to write a temporary file.
Please allow for objects or string

**Comment by tlivings** [comment]
This module is built on the premise of a design-driven development workflow. How are you generating your api document?

**Comment by tlivings** [comment]
In addition, objects can not have circular schemas, whereas with `$ref` an api spec may.

**Comment by tlivings** [comment]
I am reopening this to think about more.

**Comment by dcharbonnier** [comment]
I'm generating my documents with a keyboard and first, that's why I use and like swagguerize-hapi ;-) But I separate the different domains of the api, I also separate the data models and the path models.
It's easier to read, easier to conceive, to not introduce unnecessary dependencies, easier to add, remove a part of the api.
When the server start I just collect all the data, all the handlers, I merge them, I validate the schema and I pass it to swaggerize hapi. This could be a build step or as I did with this new version write a tmp file with the content but I don't see the point and the previous version was just more flexible.
It seams more easy to await a readfile on the config options than to write a tmp file just because the module only support a file path.

Exemple:
```
├── authorizations
│   ├── index.ts
│   ├── models
│   │   ├── Client.ts
│   │   ├── index.ts
│   │   └── User.ts
│   ├── rest
│   │   ├── handlers
│   │   │   ├── clients
│   │   │   │   └── {id}.ts
│   │   │   ├── clients.ts
│   │   │   ├── users
│   │   │   │   ├── {id}
│   │   │   │   │   └── token.ts
│   │   │   │   └── {id}.ts
│   │   │   └── users.ts
│   │   └── swagger
│   │       ├── definitions.yaml
│   │       └── paths.yaml
│   ├── scope.ts
│   ├── Signature.ts
│   └── tests
│       ├── Authentication.spec.ts
│       ├── Client.spec.ts
│       ├── index.ts
│       ├── User.spec.ts
│       └── utils.ts
│── xxx
│   ├── xx.ts
│   ├── xx.ts
│   ├── Host.ts
│   ├── models
│   │   └── xxx.ts
│   ├── rest
│   │   ├── handlers
│   │   │   └── xxx
│   │   │       └── {xxx}
│   │   │           └── {xxx}.ts
│   │   └── swagger
│   │       └── paths.yaml
│   ├── xxx.ts
│   └── tests
│       ├── xxx.spec.ts
│       ├── xxx.spec.ts
│       └── xxx.spec.ts

```

**Comment by tlivings** [comment]
Added in #113. Published version 4.1.0.

Note: you do not need to validate the API, swaggerize-hapi will do that for you.
